### ISSUE TITLE
configure: WARNING: unrecognized options: --with-configfile

### ISSUE THREAD

**Issue by ncp1113** [mikebrady/shairport-sync#440]
This is possibly the same issue as #409, which is closed. This is the command I used for configure
./configure --with-alsa --with-stdout --with-pipe --with-avahi --with-ssl=openssl --with-soxr --with-metadata --with-systemd --with-configfile

and I received:
configure: WARNING: unrecognized options: --with-configfile

Without that the program starts at machine startup but it doesn't read the config file that I copied from the repo and put in /etc/shairport-sync.conf

I am using a raspberry 3 with the most recent version of rasping
Linux raspberrypi 4.4.37-v7+ #936 SMP Fri Dec 9 16:56:49 GMT 2016 armv7l GNU/Linux

The full output from configure is:
pi@raspberrypi:~/shairport-sync$ ./configure --with-alsa --with-stdout --with-pipe --with-avahi --with-ssl=openssl --with-soxr --with-metadata --with-systemd --with-configfile
configure: WARNING: unrecognized options: --with-configfile
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for style of include used by make... GNU
checking dependency style of gcc... gcc3
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for clock_gettime in -lrt... yes
>>Including libdaemon
checking for DAEMON... yes
checking for pthread_create in -lpthread... yes
checking for exp in -lm... yes
>>Including libpopt
checking for POPT... yes
>>Including the stdout audio back end
>>Including the pipe audio back end
checking for LIBCONFIG... yes
checking encryption libraries chosen... checking for SSL... yes
>>Including support for soxr-based interpolation
checking for soxr_create in -lsoxr... yes
>>Including metadata support
>>Including Avahi mDNS support
checking for avahi_client_new in -lavahi-client... yes
checking for avahi_strerror in -lavahi-common... yes
>>Including an ALSA back end
checking for ALSA... yes
checking for xmltoman... no
configure: WARNING: >>xmltoman not found - not rebuilding man pages
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking getopt_long.h usability... no
checking getopt_long.h presence... no
checking for getopt_long.h... no
checking arpa/inet.h usability... yes
checking arpa/inet.h presence... yes
checking for arpa/inet.h... yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking limits.h usability... yes
checking limits.h presence... yes
checking for limits.h... yes
checking mach/mach.h usability... no
checking mach/mach.h presence... no
checking for mach/mach.h... no
checking for memory.h... (cached) yes
checking netdb.h usability... yes
checking netdb.h presence... yes
checking for netdb.h... yes
checking netinet/in.h usability... yes
checking netinet/in.h presence... yes
checking for netinet/in.h... yes
checking for stdint.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking sys/ioctl.h usability... yes
checking sys/ioctl.h presence... yes
checking for sys/ioctl.h... yes
checking sys/socket.h usability... yes
checking sys/socket.h presence... yes
checking for sys/socket.h... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking syslog.h usability... yes
checking syslog.h presence... yes
checking for syslog.h... yes
checking for unistd.h... (cached) yes
checking for inline... inline
checking for int16_t... yes
checking for int32_t... yes
checking for int64_t... yes
checking for pid_t... yes
checking for size_t... yes
checking for ssize_t... yes
checking for uint16_t... yes
checking for uint32_t... yes
checking for uint64_t... yes
checking for uint8_t... yes
checking for working alloca.h... yes
checking for alloca... yes
checking for error_at_line... yes
checking vfork.h usability... no
checking vfork.h presence... no
checking for vfork.h... no
checking for fork... yes
checking for vfork... yes
checking for working fork... yes
checking for working vfork... (cached) yes
checking for stdlib.h... (cached) yes
checking for GNU libc compatible malloc... yes
checking for stdlib.h... (cached) yes
checking for GNU libc compatible realloc... yes
checking for atexit... yes
checking for clock_gettime... yes
checking for gethostname... yes
checking for inet_ntoa... yes
checking for memchr... yes
checking for memmove... yes
checking for memset... yes
checking for mkfifo... yes
checking for pow... yes
checking for select... yes
checking for socket... yes
checking for stpcpy... yes
checking for strcasecmp... yes
checking for strchr... yes
checking for strdup... yes
checking for strerror... yes
checking for strstr... yes
checking for strtol... yes
checking for strtoul... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating man/Makefile
config.status: creating scripts/shairport-sync.service
config.status: creating scripts/shairport-sync
config.status: creating config.h
config.status: config.h is unchanged
config.status: executing depfiles commands
configure: WARNING: unrecognized options: --with-configfile

**Comment by mikebrady** [comment]
Thanks for the post. The option `--with-configfile` was removed a long time ago, and now Shairport Sync always looks for a configuration file. I think the problem you are having is due to the new-ish behaviour of Shairport Sync, which is that by default it now looks in `/use/local/etc/` for the configuration file. To compile it so that it looks in `/etc/`, use the configuration option `--sysconfdir=/etc`. BTW, the change was made to make the Shairport Sync easier to package.

**Comment by ncp1113** [comment]
Thank you, Mike! I simply moved the config file that I already edited to /usr/share/etc and rebooted.  Now it is reading the config.

I will try to update the documentation.

Nancy

**Comment by ncp1113** [comment]
I looked at the README.md file and it looks like it has already been changed.  The line (142)

- `--with-` to install a configuration file and a separate sample file at the `make install` stage. Default is to install. An existing `/etc/shairport-sync.conf` will not be overwritten.
 Looks like it should be removed and this line should be added:
- `-sysconfdir=/etc`  To install the configuration file to the legacy location /etc. The new default is to install the configuration file to /usr/local/etc

The beginning of  line (237) should be changed to:
Shairport Sync reads settings from a configuration file at `/usr/local/etc/shairport-sync.conf`.

Nancy
